src/firefly/js/templates/router/RouteHelper.jsx

```diff
@@ -94,7 +94,7 @@ function useFormWatcher(submitTo='/?results', onCancel) {
                 }
             });
         });
-    },[]);
+    },[submitTo]);
```

Comment:
@loitly could I request your feedback on this? it's related to this change in the IFE PR in [SourceInspector.jsx](https://github.com/IPAC-SW/irsa-ife/pull/326/files#diff-3817738aa78c72dcf422585b88c6add4bc06ea705344aa9ac81d079485931b5bR166-R168): 

`Search by ID` results can look like either `Region Explorer` or `Source Inspector `results. So I needed to tweak the call to `FormWatcher's` `submitTo` function so that it can change based on user's selection. 

But for this to work, I needed to add `submitTo` as a dependency to the `useEffect` in the `useFormWatcher` hook. Let me know if that makes sense.

Reply:
https://github.com/Caltech-IPAC/firefly/blob/d91eb076a95139c9ce47a1e4fbcc5be5de0951b6/src/firefly/js/templates/router/RouteHelper.jsx#L60

`submitTo` can be a path(string) or a function that return a path.  So technically, you can achieve what you want to do without any changes.  However, if you want to add this, be aware that `FormWatcher` will re-render when a new `submitTo` is passed.

Also, I noticed this code is missing cleanup.  Please add `return (() => dispatchCancelActionWatcher(id));` at the end of the `useEffect`.

Reply:
Added the cleanup for the `useEffect`... about the dependency, the re-render of `FormWatcher` is intentional and is what I was trying to achieve. 

Reason: I am  passing in `submitTo` as a string. But it still needs to change and re-render `FormWatcher` when the user 'changes' this string (via selecting `tile` vs `object` id). Because otherwise, since `FormWatcher` component is already called with previous value of `submitTo`, a re-render does not happen when the user changes their selection (since `submitTo` was not a dependency of the `useEffect`) and they end up going to the wrong path. Re-render makes sure it gets the latest value of `submitTo`...

Let me know if that sounds good?